python/ray/autoscaler/_private/kuberay/node_provider.py

```diff
@@ -191,6 +191,33 @@ def _get(self, path: str) -> Dict[str, Any]:
         assert result.status_code == 200
         return result.json()
 
+    def _get_non_terminating_pods(
+        self, tag_filters: Dict[str, str]
+    ) -> List[Dict[str, Any]]:
+        """Get the list of pods in the Ray cluster, excluding pods
+        marked for termination.
+
+        Filter by the specified tag_filters.
+
+        Return a list of pod objects, represented as dictionaries.
+        """
+        label_filters = to_label_selector(
+            {
+                "ray.io/cluster": self.cluster_name,
+            }
+        )
+        data = self._get("pods?labelSelector=" + requests.utils.quote(label_filters))
+        result = []
+        for pod in data["items"]:
+            if "deletionTimestamp" in pod["metadata"]:
```

Comment:
Add a comment here about where 'deletionTimestamp' comes from (essentially that it is what k8s uses)

Reply:
Added comment + link to K8s docs.